bootstrap.py

```diff
@@ -45,206 +53,118 @@
-    'snippets = spyder.plugins.completion.providers.snippets.provider:'
-    'SnippetsProvider'
-)
-
-lsp = pkg_resources.EntryPoint.parse(
-    'lsp = spyder.plugins.completion.providers.languageserver.provider:'
-    'LanguageServerProvider'
-)
-
-# Create a fake Spyder distribution
-d = pkg_resources.Distribution(__file__)
-
-# Add the providers to the fake EntryPoint
-d._ep_map = {
-    'spyder.completions': {
-        'fallback': fallback,
-        'snippets': snippets,
-        'lsp': lsp,
-    }
-}
-
-# Add the fake distribution to the global working_set
-pkg_resources.working_set.add(d, 'spyder')
-
-
 # Selecting the GUI toolkit: PyQt5 if installed
 if args.gui is None:
     try:
         import PyQt5  # analysis:ignore
-        print("*. PyQt5 is detected, selecting")
+        logger.info("PyQt5 is detected, selecting")
         os.environ['QT_API'] = 'pyqt5'
     except ImportError:
         sys.exit("ERROR: No PyQt5 detected!")
 else:
-    print("*. Skipping GUI toolkit detection")
+    logger.info("Skipping GUI toolkit detection")
     os.environ['QT_API'] = args.gui
 
+# ---- Install sub repos
+
+if not args.no_install:
+    for name in REPOS.keys():
+        if not REPOS[name]['editable']:
+            install_repo(name)
+        else:
+            logger.info("%s already installed in editable mode", name)
+
+# ---- Check versions
 
 # Checking versions (among other things, this has the effect of setting the
 # QT_API environment variable if this has not yet been done just above)
 from spyder import get_versions
 versions = get_versions(reporev=True)
-print("*. Imported Spyder %s - Revision %s, Branch: %s" %
-      (versions['spyder'], versions['revision'], versions['branch']))
-print("    [Python %s %dbits, Qt %s, %s %s on %s]" %
-      (versions['python'], versions['bitness'], versions['qt'],
-       versions['qt_api'], versions['qt_api_ver'], versions['system']))
-
+logger.info("Imported Spyder %s - Revision %s, Branch: %s; "
+            "[Python %s %dbits, Qt %s, %s %s on %s]",
+            versions['spyder'], versions['revision'], versions['branch'],
+            versions['python'], versions['bitness'], versions['qt'],
+            versions['qt_api'], versions['qt_api_ver'], versions['system'])
 
 # Check that we have the right qtpy version
 from spyder.utils import programs
 if not programs.is_module_installed('qtpy', '>=1.1.0'):
-    print("")
     sys.exit("ERROR: Your qtpy version is outdated. Please install qtpy "
              "1.1.0 or higher to be able to work with Spyder!")
 
-
-# --- Executing Spyder
+# ---- Execute Spyder
 
 if args.show_console:
-    print("(Deprecated) --show console does nothing, now the default behavior "
-          "is to show the console, use --hide-console if you want to hide it")
+    logger.info("(Deprecated) --show console does nothing, now the default "
+                "behavior is to show the console, use --hide-console if you "
+                "want to hide it")
```

Comment:
You removed the `show_console` option above, so this needs to go as well, I'd say.